vendor/src/github.com/Sirupsen/logrus/writer.go

```diff
@@ -7,18 +7,40 @@ import (
 )
 
 func (logger *Logger) Writer() *io.PipeWriter {
+	return logger.WriterLevel(InfoLevel)
+}
+
+func (logger *Logger) WriterLevel(level Level) *io.PipeWriter {
 	reader, writer := io.Pipe()
 
-	go logger.writerScanner(reader)
+	var printFunc func(args ...interface{})
+	switch level {
```

Comment:
How did these end up "re-leveled"? Seems odd to be doing dispatch at this point. How was this handled before?